File: csharp/prime-factors/PrimeFactors.cs
using System;
using System.Linq;
using System.Collections.Generic;

public static class PrimeFactors
{
    public static int[] Factors(long number)
    {
        var result = new List<int>();
        var possible = 2;
        
        while(number > 1 && possible <= number)
        {
            if(number % possible != 0)
                possible += 1;
            else
            {
                result.Add(possible);
                number = number / possible;
            }
        }

        return result.ToArray();
    }
}

File: go/prime-factors/prime_factors.go
package prime

func Factors(n int64) []int64 {
	var i int64 = 2
	for i <= n {
		if n%i == 0 {
			return append([]int64{i}, Factors(n/i)...)
		}
		i++
	}
	return []int64{}
}
